File: projektvorlage/python/atoms/_template/example/main_test.py
"""Test des Atoms nebenan. Laeuft unveraendert gruen — erst ersetzen, dann bauen."""

import pytest

from atoms._template.example.main import run


def test_gibt_die_eingabe_zurueck():
    assert run(3) == 3


def test_weist_falschen_typ_ab():
    with pytest.raises(TypeError):
        run("3")

File: projektvorlage/typescript/atoms/_template/example/main_test.ts
// Test des Atoms nebenan. Laeuft unveraendert gruen — erst ersetzen, dann bauen.

import { test } from "node:test";
import assert from "node:assert/strict";

import { run } from "./main.ts";

test("gibt die eingabe zurueck", () => {
  assert.equal(run(3), 3);
});

test("weist falschen typ ab", () => {
  assert.throws(() => run("3" as unknown as number), TypeError);
});
